### PR TITLE
Add Section Filters to Submissions Page

### PR DESCRIPTION
**Pull request by alexinman**
Added a secondary filter option to the React table class.
Used this new secondary filter to add a filter for sections to the Submissions page.
Changed the Collect [All | Section] Submissions button to just be a Collect Submissions button which collects the selected submissions.
This allows users to filter by section, click the select all and then collect submissions in order to collect a single section's submissions while also allowing them to collect all submissions by just not filtering by section. This functionality also allows users to collect any combination of submissions from the submission page.

screen shot 2016-04-01 at 3 55 53 pm
Screenshot shows all new things:
 - The (red) error message appears when you try to collect (a) submission(s) that their due date (whether section or global) hasn't passed yet.
 - The (green) success message appears when at least one submission you selected is successfully being collected.
 - The new dropdown by the search field to filter by sections
 - The updated "Collect Submissions" button

### CODE REVIEW

File: app/assets/javascripts/ReactComponents/table.js [hunk 1]
```diff
@@ -169,7 +197,16 @@ var Table = React.createClass({displayName: 'Table',
     } else {
       columns = this.props.columns;
     }
-
+    var secondary_filter_div = null;
+    if(this.props.secondary_filters != null){
```
Comment (david-yz-liu) [inline app/assets/javascripts/ReactComponents/table.js]: Use whitespace. The style is off in this entire block.

File: app/controllers/submissions_controller.rb [hunk 2]
```diff
@@ -232,42 +232,38 @@ def collect_all_submissions
...
-    successes = Array.new
-    noSubmissions = Array.new
-    section_ids.each do |id|
-      if id == '0'
-        submission_collector = SubmissionCollector.instance
-        submission_collector.push_groupings_to_queue(
-          assignment.sectionless_groupings)
-        successes.push(t('groups.unassigned_students'))
+    groupings = assignment.groupings.find(params[:groupings])
+    groupings_to_be_collected = Array.new
```
Comment (david-yz-liu) [inline app/controllers/submissions_controller.rb]: Use http://ruby-doc.org/core-2.3.0/Enumerable.html#method-i-partition

File: app/views/submissions/_submissions_table.js.jsx.erb [hunk 3]
```diff
@@ -195,6 +195,24 @@
       // Do student-specific table stuff here.
       // possibly move this to getdefaultprops.
 
+      // need to filter for each section
+      var sections = <%= raw @sections.to_json %>;
+      if(sections.length > 0){
```
Comment (david-yz-liu) [inline app/views/submissions/_submissions_table.js.jsx.erb]: Whitespace

File: app/views/submissions/_submissions_table.js.jsx.erb [hunk 4]
```diff
@@ -195,6 +195,24 @@
...
+        var section_filters = [];
+        var all = '<%= I18n.t('browse_submissions.sections.all') %>';
+        section_filters.push({name: all, text: all, func: function(g) { return g; } });
+        for (var i = 0; i < sections.length; i++) {
+          var filter = {
+            name: sections[i].name,
+            text: sections[i].name,
+            func: function(submission) {
+              return (submission.section == this.name);
```
Comment (david-yz-liu) [inline app/views/submissions/_submissions_table.js.jsx.erb]: Use `===`

File: app/views/submissions/_submissions_table.js.jsx.erb [hunk 5]
```diff
@@ -307,31 +326,13 @@
...
-      });
-    },
-    collectSectionSubmissions: function() {
+    collectSubmissions: function() {
       var dataLoad = {
-        sections: this.refs.sectionSelect.state.value.split(',')
+        groupings: this.props.selected_submissions
       };
       jQuery.ajax({
         method: 'GET',
```
Comment (david-yz-liu) [inline app/views/submissions/_submissions_table.js.jsx.erb]: I didn't think of this before, but please make this a `POST` request. The data will sometimes be too big to send in a header.

**Comment by david-yz-liu** [comment]
@alexinman nice work. Having a set of secondary filters doesn't strike me as particularly elegant, but it suits this purpose fine enough.

File: app/views/submissions/_submissions_table.js.jsx.erb [hunk 6]
```diff
@@ -368,6 +369,14 @@
     render: function() {
       return (
         <div className='react-release-marks'>
+<<<<<<< HEAD
```
Comment (david-yz-liu) [inline app/views/submissions/_submissions_table.js.jsx.erb]: @alexinman

Comment (alexinman) [inline app/views/submissions/_submissions_table.js.jsx.erb]: Missed this one. Fixing it now...

**Pull request merged**